public/app/features/dashboard/state/DashboardModel.ts

```diff
@@ -480,6 +480,11 @@ export class DashboardModel implements TimeModel {
     this.events.publish(new DashboardPanelsChangedEvent());
   }
 
+  updateMeta(updates: Partial<DashboardMeta>) {
+    this.meta = { ...this.meta, ...updates };
+    this.events.publish(new DashboardMetaChangedEvent());
+  }
```

Comment:
@owensmallwood so this bit here is a mutation of the dashboard instance that is passed around everywhere via DashboardPage, but changing a property on this instance does not trigger a react update anywhere. 

Part of the reason here is that DashboardModel comes before Grafana was written in React and we supported many angular panels and services that depended on this instance staying the same (until you change dashboard)

Reply:
Thanks for the background and context around that. Makes sense!